client/http/http.go

```diff
@@ -56,18 +56,28 @@ func NewWithInfo(url string, info *chain.Info, transport nhttp.RoundTripper) (cl
 func ForURLs(urls []string, chainHash []byte) []client.Client {
 	clients := make([]client.Client, 0)
 	var info *chain.Info
+	skipped := []string{}
 	for _, u := range urls {
 		if info == nil {
 			if c, err := New(u, chainHash, nil); err == nil {
 				info, err = c.Info(context.Background())
 				clients = append(clients, c)
+			} else {
+				skipped = append(skipped, u)
 			}
 		} else {
 			if c, err := NewWithInfo(u, info, nil); err == nil {
 				clients = append(clients, c)
 			}
 		}
 	}
+	if info != nil {
```

Comment:
But info can be nil here (which is allowed by the code at my previous comment). Then you would not add the skipped clients to the result, but you would still returned the other clients? Even they all failed on the call to c.Info?

Reply:
technically that's true.  The concrete implementation of New / Info currently only have one failure mode - when Info fails to load. If it remains nil after the loop, it means its failed to load from all URLs, and an empty array of clients will be returned as a signal of failure..